### PR TITLE
Docs

### PR DESCRIPTION
**Pull request by almarklein**
Prerequisite for https://github.com/plotly/dash-docs/pull/1039

* Tweaked the docstrings in the VolumeSlicer to be more usable in markdown. 
* Added a docsting at the top of the module that defines VolumeSlicer, to specify additional reference docs.
* Added a new `docs` module with a few simple utilities:
  * A function to get the reference docs as markdown. -> To be used in dash-docs.
  * A function to add the reference docs to the readme. This function is called when the file is executed as a script.
* Added tests for the above.
* General improvements to the docs.

### CODE REVIEW

File: dash_slicer/docs.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+
+def get_reference_docs():
+    """Create reference documentation from the source code.
+    A bit like Sphinx autodoc, but using Markdown, and more basic.
+    Returns a str in Markdown format.
+
+    Note that this function is used to build the Dash Slicer chapter
+    in the Dash docs.
+    """
```
Comment (almarklein) [inline dash_slicer/docs.py:37]: I'm not really sure whether to put his here or in dash-docs. The only real advantage is that we can also add our reference docs to the readme, but perhaps we should just provide a link to `https://dash.plotly.com/slicer`

Comment (almarklein) [reply dash_slicer/docs.py:37]: Thoughts welcome :)

**Comment by almarklein** [comment]
(I know what to do about the Windows CI error, but I'll wait with a fix until we figure out how to move forward)

File: README.md [hunk 2]
```diff
@@ -54,7 +54,131 @@ This code is distributed under MIT license.
...
+  red, green, or blue, depending on the axis. Set to empty string
+  for "no color".
+* `thumbnail` (`int` or `bool`): linear size of low-resolution data to be
+  uploaded to the client. If `False`, the full-resolution data are
+  uploaded client-side. If `True` (default), a default value of 32 is
+  used.
+
+Note that this is not a Dash component. The components that make
+up the slicer (and which must be present in the layout) are:
+`slicer.graph`, `slicer.slider`, and `slicer.stores`.
```
Comment (emmanuelle) [inline README.md:98]:
```suggestion
`slicer.graph`, `slicer.slider` (optional in layout if several slicers are used), and `slicer.stores`.
```

File: README.md [hunk 3]
```diff
@@ -54,7 +54,131 @@ This code is distributed under MIT license.
...
+**property `VolumeSlicer.stores`**: A list of `dcc.Store` objects that the slicer needs to work.
+These must be added to the app layout.
+
+
+
+### Reacting to slicer state
+
+It is possible to get notified of updates to slicer position and
+view ranges. To get this for all slicers with a specific scene_id, create
+a pattern matching input like this:
```
Comment (emmanuelle) [inline README.md:149]:
```suggestion
a [pattern matching input](https://dash.plotly.com/pattern-matching-callbacks) like this:
```

File: dash_slicer/docs.py (hunk 1 above)
Comment (emmanuelle) [reply dash_slicer/docs.py:37]: I think it's a good idea to have it here if in dash-docs you can just call this function.

**Review by emmanuelle — APPROVED** [review]

**Comment by emmanuelle** [comment]
I approved a bit too fast (there's the Windows CI problem you mentioned), just wanted to say that the approach is sound I think.

File: README.md (hunk 2 above)
Comment (almarklein) [reply README.md:98]: Unfortunately it is not optional, because we have it as input and output to our callbacks.

**Pull request merged**